src/thor/isochrone.cc

```diff
@@ -17,10 +17,22 @@ constexpr float METRIC_PADDING = 10.f;
 template <typename PrecisionT>
 std::vector<GeoPoint<PrecisionT>> OriginEdgeShape(const std::vector<GeoPoint<PrecisionT>>& pts,
                                                   double distance_along) {
+  // just the endpoint really
+  if (distance_along == 0)
+    return {pts.back(), pts.back()};
+
+  // consume shape until we reach the desired distance
   double suffix_len = 0;
   for (auto from = std::next(pts.rbegin()), to = pts.rbegin(); from != pts.rend(); ++from, ++to) {
+    // add whatever this segment of shape contributes to the overall distance
     PrecisionT len = from->Distance(*to);
     suffix_len += len;
+
+    // skip 0 length segments
+    if (len == 0)
+      continue;
```

Comment:
this is not unneeded. if `len` is 0 then it cant change `suffix_len` which means it cant change the outcome of that `if` below and get in there and do a divide by 0.